spec/models/program_session_spec.rb

```diff
@@ -1,5 +1,99 @@
 require 'rails_helper'
 
-RSpec.describe ProgramSession, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+describe ProgramSession do
+
+  let(:proposal) { create(:proposal, :with_two_speakers) }
+
+  it "responds to .create_from_proposal" do
+    expect(ProgramSession).to respond_to(:create_from_proposal)
+  end
+
+  describe ".create_from_proposal" do
+    context "creates a program session from a proposal" do
```

Comment:
Not sure this context is necessary. I tend to think of context blocks as representing a particular state. Like, "user is an organizer" or "event has no end date set".